packages/core/src/tools/errorFilter.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { Configuration } from '../domain/configuration'
+import { mockClock } from '../../../core/test/specHelper'
+import { RawError } from './error'
+import { createErrorFilter, ErrorFilter } from './errorFilter'
+import { RelativeTime, relativeToClocks, resetNavigationStart } from './timeUtils'
+import { noop, ONE_MINUTE } from './utils'
+
+const CONFIGURATION = { maxErrorsByMinute: 1 } as Configuration
+
+describe('errorFilter.isLimitReached', () => {
+  let errorFilter: ErrorFilter | undefined
+  let clock: jasmine.Clock
+  let cleanupClock: () => void
+
+  beforeEach(() => {
+    ;({ clock, stop: cleanupClock } = mockClock())
+    resetNavigationStart()
+  })
+
+  afterEach(() => {
+    cleanupClock()
+  })
+
+  it('returns false if the limit is not reached', () => {
+    errorFilter = createErrorFilter(CONFIGURATION, noop)
+
+    expect(errorFilter.isLimitReached()).toBe(false)
+  })
+
+  it('returns true if the limit is reached', () => {
+    errorFilter = createErrorFilter(CONFIGURATION, noop)
+
+    errorFilter.isLimitReached()
+    expect(errorFilter.isLimitReached()).toBe(true)
+  })
+
+  it('returns false again when one minute is passed after the first counted error', () => {
+    errorFilter = createErrorFilter(CONFIGURATION, noop)
+
+    errorFilter.isLimitReached()
+    errorFilter.isLimitReached()
+    clock.tick(ONE_MINUTE)
+    expect(errorFilter.isLimitReached()).toBe(false)
+  })
+
+  it('calls the "onLimitReached" callback with the raw "limit reached" error to send when the limit is reached', () => {
+    const onLimitReachedSpy = jasmine.createSpy<(rawError: RawError) => void>()
+    errorFilter = createErrorFilter(CONFIGURATION, onLimitReachedSpy)
+
+    errorFilter.isLimitReached()
+    errorFilter.isLimitReached()
+    expect(onLimitReachedSpy).toHaveBeenCalledOnceWith({
+      message: 'Reached max number of errors by minute: 1',
+      source: 'agent',
+      startClocks: relativeToClocks(0 as RelativeTime),
+    })
+  })
+
+  it('returns false while calling the "onLimitReached" to allow sending the "limit reached" error', () => {
```

Comment:
FWIW, I think `shouldSendError` was a better name. Sure, `isLimitReached` reflects the internal state of the filter a bit better, but it is not always right (since when the callback is called, the limit *is* reached but `isLimitReached` returns `false`). `shouldSendError` is more abstract and related to the end-usage of the filter, but it is more accurate.

Reply:
IMO, we could have an alternate naming that:
- is not tied to the limit
- is not tied to the end usage 
something like `errorFilter.shouldAllowError()`

Reply:
Let's roll with `isLimitReached`, it's explicit enough.